devito/arch/compiler.py

```diff
@@ -581,10 +586,14 @@ class CustomCompiler(Compiler):
     """
 
     def __new__(cls, *args, **kwargs):
+        platform = kwargs.pop('platform', configuration['platform'])
+
         if any(i in environ for i in ['CC', 'CXX', 'CFLAGS', 'LDFLAGS']):
             obj = super().__new__(cls, *args, **kwargs)
             obj.__init__(*args, **kwargs)
             return obj
+        elif platform is M1:
```

Comment:
I'd probably put it as first `if` in the unlikely case it's an M! but someone declared CC